fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java

```diff
@@ -589,9 +589,15 @@ private SlotReference getContextPartitionColumn(IncrementCheckerContext context)
         private static boolean checkPartition(Collection<? extends Expression> expressionsToCheck, Plan plan,
                 IncrementCheckerContext context) {
             NamedExpression partitionColumn = context.getMvPartitionColumn();
-            for (Expression projectSlot : expressionsToCheck) {
+
+            OUTER_CHECK: for (Expression projectSlot : expressionsToCheck) {
                 if (projectSlot.isColumnFromTable() && projectSlot.equals(partitionColumn.toSlot())) {
-                    continue;
+                    if (!context.getPartitionExpression().isPresent()) {
```

Comment:
if `projectSlot.isColumnFromTable() && projectSlot.equals(partitionColumn.toSlot()) ` is true
the partitionExpression in context is always right because line 667. the code is duplicated to line 667.

if `return context.getMvPartitionColumn().isColumnFromTable();` at end. this can be removed.

Reply:
line 667 may not execute.